ocs_ci/cleanup/aws/cleanup.py

```diff
@@ -35,7 +38,35 @@ def cleanup(cluster_name, cluster_id):
     run_cmd(f"{oc_bin} destroy cluster --dir {cleanup_path} --log-level=debug")
 
 
-def main():
+def get_clusters_to_delete(time_to_delete):
+    """
+    Get all cluster names that their EC2 instances running time is greater
+    than the specified time to delete
+
+    Args:
+         time_to_delete (int): The maximum time in seconds that is allowed
+            for clusters to continue running
+
+    Returns:
+        list: The cluster names (e.g ebenahar-cluster-gqtd4) to be provided to the
+            ci-cleanup script
+
+    """
+    conn = boto3.resource('ec2', region_name='us-east-2')
+    instances = conn.instances.filter(Filters=[{'Name': 'tag:Name', 'Values': ['*master-0']}])
```

Comment:
I think we should select clusters for delete from VPCs as for each cluster we have one VPC. From VPC you can read tag:
 kubernetes.io/cluster/CLUSTER-name-ktzdr   where ktzdr is auto generated suffix.  But most  resources created with cluster should be tagged with such tag kubernetes.io/cluster/CLUSTER-name-ktzdr so you can find all ec2 instances for such cluster by that tag.

Haven't checked all the data provided from VPC object but I guess there will be also some info about creation datetime so you can directly take from vpc maybe.

Reply:
done